File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
import math


def solution(n, words):
    ans = [0, 0]
    prev_word = words[0]
    visited = set([prev_word])
    for index, word in enumerate(words[1:]):
        if prev_word[-1] != word[0] or word in visited:
            return [(index + 1) % n + 1,  math.ceil((index + 2) / n)]
        prev_word = word
        visited.add(word)
    
    
    return ans

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
const solution = (n, words) => {
  const wordsObj = {};
  wordsObj[words[0]] = 1;

  for (let i = 1; i < words.length; i++) {
    if (
      wordsObj[words[i]] ||
      words[i - 1][words[i - 1].length - 1] !== words[i][0]
    ) {
      return [(i % n) + 1, Math.floor(i / n) + 1];
    }
    wordsObj[words[i]] = 1;
  }
  return [0, 0];
};
